listener/upload_listener.py

```diff
@@ -373,38 +379,50 @@ def process_message(msg):
             MESSAGE_PARSE_ERROR.inc()
             LOGGER.exception("Unable to parse message: ")
             return
+        # send message to payload tracker
+        send_msg_to_payload_tracker(msg_dict, 'received', loop=loop)
 
         if msg.topic == mqueue.UPLOAD_TOPIC:
             # proces only archives from smart_management accounts
             identity = get_identity(msg_dict.get("b64_identity", ""))
             if identity is None:
                 INVALID_IDENTITY.inc()
-                LOGGER.warning("Skipped upload due to invalid identity header.")
+                error_msg = "Skipped upload due to invalid identity header."
+                LOGGER.warning(error_msg)
+                send_msg_to_payload_tracker(msg_dict, 'error', status_msg=error_msg, loop=loop)
                 return
             if not is_entitled_smart_management(identity, allow_missing_section=True):
                 MISSING_SMART_MANAGEMENT.inc()
-                LOGGER.debug("Skipped upload due to missing smart_management entitlement.")
+                error_msg = "Skipped upload due to missing smart_management entitlement."
+                LOGGER.debug(error_msg)
+                send_msg_to_payload_tracker(msg_dict, 'error', status_msg=error_msg, loop=loop)
                 return
             process_func = process_upload
         elif msg.topic == mqueue.EVENTS_TOPIC:
             if msg_dict['type'] == 'delete':
                 process_func = process_delete
+                send_msg_to_payload_tracker(msg_dict, 'processing', loop=loop)
             else:
                 UNKNOWN_EVENT_TYPE.inc()
                 LOGGER.error("Received unknown event type: %s", msg_dict['type'])
+                send_msg_to_payload_tracker(msg_dict, 'error', status_msg="Received unknown event type: %s" % msg_dict['type'], loop=loop)
                 return
         else:
             UNKNOWN_TOPIC.inc()
             LOGGER.error("Received message on unsupported topic: %s", msg.topic)
+            send_msg_to_payload_tracker(msg_dict, 'error', status_msg="Received message on unsupported topic: %s" % msg.topic, loop=loop)
             return
 
         if 'id' not in msg_dict or msg_dict["id"] is None:
             MISSING_ID.inc()
-            LOGGER.warning("Unable to process message, inventory ID is missing.")
+            error_msg = "Unable to process message, inventory ID is missing."
+            LOGGER.warning(error_msg)
+            send_msg_to_payload_tracker(msg_dict, 'error', status_msg=error_msg, loop=loop)
             return
 
         future = executor.submit(process_func, msg_dict, loop=loop)
         future.add_done_callback(on_thread_done)
+        send_msg_to_payload_tracker(msg_dict, 'success', loop=loop)
```

Comment:
this should not success here because the payload is going to evaluator yet

should success here only if archive hasn't changed and is not going to evaluator